File: Data-Structures-and-Algorithms-using-Swift/Sorting/InsertionSort.swift
import Foundation

func greater(_ value1 : Int, _ value2 : Int) -> Bool {
    return value1 > value2
}

func insertionSort(_ arr : inout [Int]) {
    let size : Int = arr.count
    var temp : Int, j : Int, i : Int = 1
    while (i < size) {
        temp = arr[i]
        j = i
        while (j > 0 && greater(arr[j - 1],temp)) {
            arr[j] = arr[j - 1]
            j -= 1
        }
        arr[j] = temp
        i += 1
    }
}

// Testing Code.
var array : [Int] = [9, 1, 8, 2, 7, 3, 6, 4, 5]
insertionSort( &array)
print(array)

// [1, 2, 3, 4, 5, 6, 7, 8, 9]

File: Data-Structures-Algorithms-In-TS/Sorting/InsertionSort.ts
function more(value1: number, value2: number): boolean {
    return value1 > value2;
}

function InsertionSort(arr: Array<number>) {
    let size: number = arr.length;
    let temp: number;
    let j: number;
    for (let i: number = 1; i < size; i++) {
        temp = arr[i];
        for (j = i; j > 0 && more(arr[j - 1], temp); j--) {
            arr[j] = arr[j - 1];
        }
        arr[j] = temp;
    }
}

// Testing code.
let array: Array<number> = [9, 1, 8, 2, 7, 3, 6, 4, 5];
InsertionSort(array);
console.log(array);
